ricecooker/classes/files.py

```diff
@@ -207,28 +207,43 @@ class ThumbnailFile(ThumbnailPresetMixin, DownloadFile):
     default_ext = file_formats.PNG
 
     def validate(self):
-        assert os.path.splitext(self.path)[1][1:] in [file_formats.JPG, file_formats.JPEG, file_formats.PNG], "Thumbnails must be in jpg, jpeg, or png format"
+        super(ThumbnailFile, self).validate()
+        if os.path.splitext(self.path)[1][1:] != "":
+            assert os.path.splitext(self.path)[1][1:] in [file_formats.JPG, file_formats.JPEG, file_formats.PNG], "Thumbnails must be in jpg, jpeg, or png format"
+
 
 class AudioFile(DownloadFile):
+    default_ext = file_formats.MP3
+
     def get_preset(self):
         return self.preset or format_presets.AUDIO
 
     def validate(self):
-        assert self.path.endswith(file_formats.MP3), "Audio files must be in mp3 format"
+        super(AudioFile, self).validate()
+        if os.path.splitext(self.path)[1][1:] != "":
+            assert self.path.endswith(file_formats.MP3), "Audio files must be in mp3 format"
 
 class DocumentFile(DownloadFile):
+    default_ext = file_formats.PDF
+
     def get_preset(self):
         return self.preset or format_presets.DOCUMENT
 
     def validate(self):
-        assert self.path.endswith(file_formats.PDF), "Document files must be in pdf format"
+        super(DocumentFile, self).validate()
+        if os.path.splitext(self.path)[1][1:] != "":
+            assert self.path.endswith(file_formats.PDF), "Document files must be in pdf format"
 
 class HTMLZipFile(DownloadFile):
+    default_ext = file_formats.HTML5
+
     def get_preset(self):
         return self.preset or format_presets.HTML5_ZIP
 
     def validate(self):
```

Comment:
These `validate` functions also all have the same general logic. Can you find a way to just have them use the base class's `validate` command?